### PR TITLE
Launch k3s on VM on Harvester

### PR DESCRIPTION
**Pull request by meysholdt**
## Description
Launch k3s in VM on Harvester.

## Related Issue(s)
Fixes #

## How to test
1. Run build with `with-vm` annotation:
```
werft run github -a with-vm=true
```

2. Log into the VM and confirm that k3s is running:
```
# Grab the Harvester kubeconfig
kubectl -n werft get secret harvester-kubeconfig -o jsonpath='{.data}' | jq -r '.["harvester-kubeconfig.yml"]' | base64 -d > harvester-kubeconfig.yml
kubectl -n werft get secret harvester-vm-ssh-keys -o jsonpath='{.data}' | jq -r '.["id_rsa"]' | base64 -d > /home/gitpod/.ssh/id_rsa
kubectl -n werft get secret harvester-vm-ssh-keys -o jsonpath='{.data}' | jq -r '.["id_rsa.pub"]' | base64 -d > /home/gitpod/.ssh/id_rsa.pub

chmod 600 /home/gitpod/.ssh/id_rsa
chmod 644 /home/gitpod/.ssh/id_rsa.pub

# Workspace: Start SSH proxy
sudo kubectl \
	--kubeconfig=harvester-kubeconfig.yml \
	-n preview-mads-harvester-k3s \
	port-forward service/proxy 22:22

# Workspace: In a new shell SSH to the VM
ssh ubuntu@127.0.0.1
```

3. Run `kubectl get pods -A ` and confirm k3s is running:
```
NAMESPACE      NAME                                       READY   STATUS      RESTARTS   AGE
cert-manager   cert-manager-57d89b9548-fpwfh              1/1     Running     0          22m
cert-manager   cert-manager-cainjector-5bcf77b697-c4vkg   1/1     Running     0          22m
cert-manager   cert-manager-startupapicheck--1-kwctf      0/1     Completed   0          22m
cert-manager   cert-manager-webhook-5f99d6f7c6-cqp8b      1/1     Running     0          22m
kube-system    calico-kube-controllers-6b9fbfff44-4h25f   1/1     Running     0          22m
kube-system    calico-node-dscxx                          1/1     Running     0          22m
kube-system    coredns-85cb69466-hq6df                    1/1     Running     0          22m
kube-system    metrics-server-5479b7cb8d-chtmc            1/1     Running     0          22m
```

This PR is the result of @ArthurSens and me pair-debugging / programming.

There is a known issue that in the Werft job, copying the k3s.yaml into the job fails because the ssh connection can not be established. See https://werft.gitpod-dev.com/job/gitpod-build-me-install-k3s.7/raw for example.  From my Gitpod workspace, I can connect via ssh to the VM without problems. 

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@meysholdt: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **loujaybee** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @loujaybee` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: .werft/vm/manifests.ts
Comment (mads-hartmann) [inline .werft/vm/manifests.ts:128]: I think we should just remove this comment - in the PR I switched from using a secret to just inlining the cloudinit, but I wasn't aware of the size restriction, so going back to a secret makes sense ☺️

File: .werft/vm/manifests.ts
Comment (mads-hartmann) [inline .werft/vm/manifests.ts:178]: I'd prefer to get rid of the alias, I don't think saving 6 chars is worth the trouble of having people figure out what `k` is ☺️

File: .werft/vm/manifests.ts
Comment (mads-hartmann) [inline .werft/vm/manifests.ts:175]: Don't we need to label the node like we did in the last cloud init?

```
kubectl label nodes ${vmName} gitpod.io/workload_meta=true gitpod.io/workload_ide=true gitpod.io/workload_workspace_services=true gitpod.io/workload_workspace_regular=true gitpod.io/workload_workspace_headless=true gitpod.io/workspace_0=true gitpod.io/workspace_1=true gitpod.io/workspace_2=true
```

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
A few tiny comments, otherwise LGTM

File: .werft/vm/manifests.ts
Comment (meysholdt) [reply .werft/vm/manifests.ts:128]: ok, removing.

File: .werft/vm/manifests.ts
Comment (meysholdt) [reply .werft/vm/manifests.ts:175]: good point. I didn't add them because I don't see them in any of the scripts in https://github.com/gitpod-io/ops/blob/main/deploy/workspace/templates/. On second theought, there the logic must be different than for us because there labels are specific to node pools. Thus adding them.

File: .werft/vm/manifests.ts
Comment (meysholdt) [reply .werft/vm/manifests.ts:178]: Can do. I'm not passionte about it.

**Comment by meysholdt** [comment]
Thx for your comments, @mads-hartmann . I've resolved them.

File: .werft/vm/manifests.ts
Comment (mads-hartmann) [inline .werft/vm/manifests.ts:171]: This is indented by a bit too much ☺️

But more importantly, vmName shouldn't be escaped, it should be interpolated by TypeScript. That also means you have to add vmName to options of the function like we do in e.g. VirtualMachineManifest ☺️

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
One last thing. Also, how do you feel about merging in @ArthurSens [PR](https://github.com/gitpod-io/gitpod/pull/7445) so we also fix the timeout once this is merged and create the missing certs namespace that was also part of the previous cloudinit ☺️

File: .werft/vm/manifests.ts
Comment (meysholdt) [reply .werft/vm/manifests.ts]: 👍 resolved.

**Review by mads-hartmann — APPROVED** [review]
Thanks 🎉

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 86e8a7ef82173bdd71cbad9f6a29db209aee1bf5</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7444#pullrequestreview-844356762" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **loujaybee** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @loujaybee` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7444#pullrequestreview-844356762" title="Approved">mads-hartmann</a>*

*No associated issue*. Requirement bypassed by manually added approval.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
